src/net_processing.cpp

```diff
@@ -1604,6 +1604,36 @@ void static ProcessGetBlockData(CNode* pfrom, const CChainParams& chainparams, c
     }
 }
 
+//! Determine whether or not a peer can request a transaction, and return a reference to it if so.
+Optional<CTransactionRef> static FindTxForGetData(CNode* peer, const CInv& inv, std::chrono::microseconds mempool_req, std::chrono::microseconds longlived_mempool_time) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
+{
+    // Check if the requested transaction is so recent that we're just
+    // about to announce it to the peer; if so, they certainly shouldn't
+    // know we already have it.
+    {
+        LOCK(peer->m_tx_relay->cs_tx_inventory);
+        if (peer->m_tx_relay->setInventoryTxToSend.count(inv.hash)) return {};
+    }
+
+    auto mi = mapRelay.find(inv.hash);
+    if (mi != mapRelay.end()) {
+        // Send stream from relay memory
+        return mi->second;
+    } else {
```

Comment:
nit: no need for this else. The if clause returns early.

Reply:
Dome, also reflowed the code a bit more.